src/api/controllers/api_key.rs

```diff
@@ -0,0 +1,190 @@
+//! # Api Key Controller
+//!
+//! Handles HTTP endpoints for api key operations including:
+//! - Create api keys
+//! - List api keys
+//! - Delete api keys
+#[cfg(feature = "authV2")]
+use crate::{
+    jobs::JobProducerTrait,
+    models::{
+        ApiError, ApiKeyModel, ApiKeyRequest, ApiKeyResponse, ApiResponse, NetworkRepoModel,
+        NotificationRepoModel, PaginationMeta, PaginationQuery, RelayerRepoModel, SignerRepoModel,
+        ThinDataAppState, TransactionRepoModel,
+    },
+    repositories::{
+        ApiKeyRepositoryTrait, NetworkRepository, PluginRepositoryTrait, RelayerRepository,
+        Repository, TransactionCounterTrait, TransactionRepository,
+    },
+};
+#[cfg(feature = "authV2")]
+use actix_web::HttpResponse;
+#[cfg(feature = "authV2")]
+use chrono::Utc;
+#[cfg(feature = "authV2")]
+use eyre::Result;
+#[cfg(feature = "authV2")]
+use uuid::Uuid;
+
+/// Create api key
+///
+/// # Arguments
+///
+/// * `api_key_request` - The api key request.
+///     * `name` - The name of the api key.
+///     * `allowed_origins` - The allowed origins for the api key.
+///     * `permissions` - The permissions for the api key.
+/// * `state` - The application state containing the api key repository.
+///
+/// # Returns
+///
+/// The result of the plugin call.
+#[cfg(feature = "authV2")]
+pub async fn create_api_key<J, RR, TR, NR, NFR, SR, TCR, PR, AKR>(
+    api_key_request: ApiKeyRequest,
+    state: ThinDataAppState<J, RR, TR, NR, NFR, SR, TCR, PR, AKR>,
+) -> Result<HttpResponse, ApiError>
+where
+    J: JobProducerTrait + Send + Sync + 'static,
+    RR: RelayerRepository + Repository<RelayerRepoModel, String> + Send + Sync + 'static,
+    TR: TransactionRepository + Repository<TransactionRepoModel, String> + Send + Sync + 'static,
+    NR: NetworkRepository + Repository<NetworkRepoModel, String> + Send + Sync + 'static,
+    NFR: Repository<NotificationRepoModel, String> + Send + Sync + 'static,
+    SR: Repository<SignerRepoModel, String> + Send + Sync + 'static,
+    TCR: TransactionCounterTrait + Send + Sync + 'static,
+    PR: PluginRepositoryTrait + Send + Sync + 'static,
+    AKR: ApiKeyRepositoryTrait + Send + Sync + 'static,
+{
+    let api_key = ApiKeyModel {
```

Comment:
maybe to create dir in models/api_key and there define files/models as:

- request
- response
- mod (domain model)
- repository (repository model)
- config (if it would be possible to use config file)

I am following this pattern with PR in draft that refactors some models to introduce crud endpoints. 
 
By having multiple models we can:

- automatically convert with try from and not manually like in this example
- use domain validations for multiple entry points (request, config*)
- use different repository model as we would need to store these keys securely(encrypted)